controllers/rider.js

```diff
@@ -215,6 +215,36 @@ const psGetPTSalaryInfo = new PS({
     `
 })
 
+const psGetRiderRatingInfo = new PS({
+    name:"rider-rating-info",
+    text: "SELECT uid, name, count(*), TRUNC(avg(value), 1) as avg_rating, date_part('year', date) as year, date_part('month', date) as month " +
+    "FROM ratings JOIN orders using(oid) JOIN riders on (riderid = uid) JOIN Users using(uid) " +
+    "GROUP BY uid, name, year, month " +
+    "ORDER BY uid, year, month;"
+});
+
+const psGetRiderSalaryInfo = new PS({
+    name:"rider-salary-summary",
+    text:"SELECT uid, name, month, year, sum(basesalary) + sum(commission) as totalSalary, sum(hoursclocked) as totalHours " +
+    "FROM receives natural join payout natural join users " +
+    "GROUP BY uid, name, year, month " +
+    "ORDER BY uid, year, month;\n"
+});
+
+const psGetRiderTotalOrdersAndAverageDeliveryTime = new PS({
+    name:"rider-total-orders-average-delivery-time",
+    text:" with Rider_Delivery_Time as (SELECT oid, date_part('year', deliveredtime) as year, " +
```

Comment:
Might need another not null check for deliveredtime

Reply:
is this handled?